### PR TITLE
Add new keyboard Dozen0

### PR DESCRIPTION
**Pull request by yynmt**
## Description
I added a new keyboard(macropad) "Dozen0".

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dozen0/config.h [hunk 1]
```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/config.h]:
```suggestion
Copyright 2019 yynmt
```

File: keyboards/dozen0/dozen0.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/dozen0.c]:
```suggestion
/* Copyright 2018 yynmt
```

File: keyboards/dozen0/dozen0.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/dozen0.h]:
```suggestion
/* Copyright 2019 yynmt
```

File: keyboards/dozen0/dozen0.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef DOZEN0_H
```
Comment (mechmerlin) [inline keyboards/dozen0/dozen0.h]: You can replace include guards with `#pragma once` for more clarity

```suggestion
#pragma once
```

File: keyboards/dozen0/dozen0.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+#define DOZEN0_H
```
Comment (mechmerlin) [inline keyboards/dozen0/dozen0.h]:
```suggestion

```

File: keyboards/dozen0/keymaps/default/config.h [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/keymaps/default/config.h]:
```suggestion
/* Copyright 2019 yynmt
```

File: keyboards/dozen0/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/keymaps/default/keymap.c]:
```suggestion
/* Copyright 2018 yynmt
```

File: keyboards/dozen0/keymaps/f12/config.h [hunk 8]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/keymaps/f12/config.h]:
```suggestion
/* Copyright 2018 yynmt
```

File: keyboards/dozen0/keymaps/f12/keymap.c [hunk 9]
```diff
@@ -0,0 +1,40 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/dozen0/keymaps/f12/keymap.c]:
```suggestion
/* Copyright 2019 yynmt
```

File: keyboards/dozen0/config.h [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+ * the keyboard. They are best used in combination with the HID Listen program,
+ * found here: https://www.pjrc.com/teensy/hid_listen.html
+ *
+ * The options below allow the magic key functionality to be changed. This is
+ * useful if your keyboard/keypad is missing keys and you want magic key support.
+ *
+ */
+
+/* key combination for magic key command */
+#define IS_COMMAND() ( \
```
Comment (drashna) [inline keyboards/dozen0/config.h]: Since this is the default, you can actually safely remove this.

File: keyboards/dozen0/rules.mk [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/dozen0/rules.mk]: Does the board have a physical reset button? 

If not, you may want to set this to lite, so there is a way to get into the bootloader. 

```suggestion
BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration(+1000)
```

Comment (yynmt) [reply keyboards/dozen0/rules.mk]: Dozen0 has a physical rest button by promicro.
The reset button is between the black spacers in front.
[Dozen0 image is here.](https://raw.githubusercontent.com/yynmt/Dozen0/master/images/main_image_mx.jpg)

**Comment by drashna** [comment]
Awesome, thanks!

**Comment by yynmt** [comment]
Thank you so much for your reviews.

**Comment by drashna** [comment]
You're very welcome! 

Thank you for adding your keyboard to the repo!

**Pull request merged**